open-api/rest-catalog-open-api.yaml

```diff
@@ -2068,6 +2162,145 @@ components:
           items:
             $ref: '#/components/schemas/PartitionStatisticsFile'
 
+    PlanTask:
+      description:
+        A JSON object that contains information provided by the server,
+        to be utilized by clients for distributed planning, should be supplied
+        as is for input in PlanTable operation.
+      type: object
+
+    FileScanTask:
+      type: object
+      required:
+        - schema
+        - spec
+        - start
+        - length
+        - data-file
+      properties:
+        data-file:
+          $ref: '#/components/schemas/ContentFile'
+        partition:
+          type: object
+          additionalProperties:
+            type: string
+        size-bytes:
+          type: number
+        start:
+          type: number
+        length:
```

Comment:
I don't think that start and length are required. The service sends back the file size in bytes and breaking files into splits should be a client-side operation. That keeps the response a bit simpler.

Reply:
From the FileScanTaskParser I think these fields seem to be required.
```
if (fileScanTask.file() != null) {
      generator.writeFieldName(DATA_FILE);
      ContentFileParser.toJson(fileScanTask.file(), spec, generator);
    }
    generator.writeNumberField(START, fileScanTask.start());
    generator.writeNumberField(LENGTH, fileScanTask.length());
```

Reply:
agree with @jackye1995 . `FileScanTaskParser` sticks with the Java implementation. If REST behavior is different, we may have to change the parser or create a new one.

Reply:
This comment seems not addressed? I agree that for scanning an entire file, length is not needed.

Reply:
Ok will remove `start` and `length`